README.md

```diff
@@ -125,6 +146,31 @@ Unstable options are not available on stable/beta with Rustfmt v1.x.
 
 See the configuration documentation on the Rustfmt [GitHub page](https://rust-lang.github.io/rustfmt/) for details (look for the `unstable_features` section).
 
+### Differences in rustfmt versions
+
+Aside differences in [configuration options](#configuring-rustfmt), versions of rustfmt may differ
+in formatting behavior.
+
+#### Default formatting of submodules
+
+On an invocation `rustfmt lib.rs`, rustfmt 1.x would format both "lib.rs" and any out-of-file
+submodules referenced in "lib.rs".
+
+With rustfmt 2.x, this behavior requires the `--recursive` flag (#3587). By default, out-of-file
+submodules of given files are not formatted.
+
+Note that this only applies to the `rustfmt` binary, and does not impact `cargo fmt`.
+
+#### Construction of config options
+
+Rustfmt 1.x uses only the configuration options declared in the rustfmt configuration file nearest
+the directory `rustfmt` is invoked.
```

Comment:
Is this strictly true in all cases? Or does it only apply in cases when multiple config files have conflicting config values, with the value from the nearest file taking precedence

Reply:
I'm not sure I understand your question. In 1.x only options in one config file would be used. In 2.x the config is constructed by merging configuration files from all parent directories with nearer directories superceding parents.

Reply:
I just wanted clarification on that, as I thought I remembered a challenge we ran into before with different config files stepping on each other. 

However, what I was thinking of was a scenario with someone running `cargo fmt` in a project that had multiple targets each with their respective config files with some different option values. The issue there though was that those different targets included similar submods, so the files for those submods were getting formatted twice with the respective config files; it wasn't a config-side issue with them interfering with each other. 

Sorry for the noise :smile: